src/HealthChecks.ApplicationStatus/ApplicationStatusHealthCheck.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+
+namespace HealthChecks.ApplicationStatus;
+
+public class ApplicationStatusHealthCheck : IHealthCheck, IDisposable
+{
+    private readonly IHostApplicationLifetime _lifetime;
+    private readonly CancellationTokenRegistration _ctRegistration;
+    private bool IsApplicationRunning { get; set; } = true;
+    public ApplicationStatusHealthCheck(IHostApplicationLifetime lifetime)
+    {
+        _lifetime = lifetime ?? throw new ArgumentNullException(nameof(IHostApplicationLifetime));
+        _ctRegistration = _lifetime.ApplicationStopping.Register(OnStopping);
+    }
+
+    private void OnStopping()
+    {
+        IsApplicationRunning = false;
```

Comment:
```suggestion
        _ctRegistration.Dispose();
        _ctRegistration = default;
```

Reply:
It should be called in the IoC itself. Since it is registered as a singleton and right before application is OFF IoC will be disposed.

Reply:
And it will be called. The `Dispose` method according to the MS recommendations should support multiple calls. It just does nothing for second and subsequent calls.